keps/sig-auth/2799-token-controller-deprecation/kep.yaml

```diff
@@ -0,0 +1,32 @@
+title: Token Controller Deprecation
+kep-number: 2799
+authors:
+  - "@zshihang"
+owning-sig: sig-auth
+participating-sigs:
+  - sig-auth
+status: implementable
+creation-date: 2021-06-25
+reviewers:
+  - "@liggitt"
+approvers:
+  - "@liggitt"
+prr-approvers:
+  - TBD
+stage: alpha
+latest-milestone: "v1.23"
+milestone:
+  alpha: "v1.23"
+feature-gates:
+  - name: TokenControllerOptOut
+    components:
+      - kube-controller-manager
+  - name: TokenControllerPurge
+    components:
+      - kube-controller-manager
+  - name: TokenControllerDeprecation
```

Comment:
I worry that this name could be confusing if we found that we had to deprecate the `TokenControllerDeprecation` feature gate.

To avoid that worry, I think I'd call it `LegacyServiceAccountTokenBehavior` and have it arrive enabled by default, then “graduate” it to `false`. This would be similar naming to `LegacyNodeRoleBehavior`.

Reply:
this feature gate is to remove TokenController from the tree. `LegacyServiceAccountTokenBehavior ` might not describe the phase correctly.